src/jit/codegenxarch.cpp

```diff
@@ -3680,6 +3680,65 @@ void CodeGen::genStructPutArgRepMovs(GenTreePutArgStk* putArgNode, unsigned base
     genConsumePutStructArgStk(putArgNode, REG_RDI, REG_RSI, REG_RCX, baseVarNum);
     instGen(INS_r_movsb);
 }
+
+//------------------------------------------------------------------------
+// If any Vector3 args are on stack and they are not pass-by-ref, the upper 32bits
+// must be cleared to zeroes. The native compiler doesn't clear the upper bits
+// and there is no way to know if the caller is native or not. So, the upper
+// 32 bits of Vector argument on stack are always cleared to zero.
+#ifdef FEATURE_SIMD
+void    CodeGen::genClearStackVec3ArgUpperBits()
+{
+#ifdef DEBUG
+    if (verbose)
+        printf("*************** In genClearStackVec3ArgUpperBits()\n");
+#endif
+
+    assert(compiler->compGeneratingProlog);
+
+    unsigned varNum = 0;
+
+    for (LclVarDsc * varDsc = compiler->lvaTable;
+    varNum < compiler->lvaCount;
+        varNum++, varDsc++)
```

Comment:
If you would prefer we don't need to increment varDsc inside for-loop

e.g.
for (unsigned varNum =0; varNum < compiler->info.compArgsCount; varNum++)
{
    LclVarDsc* varDsc = &(compiler->lvaTable[varNum]);
    assert(varDsc->lvIsParam);

    .....
}